### PR TITLE
Bela/3.11

### PR DESCRIPTION
**Pull request by argl**
Purpose and Motivation
----------------------
Make the latest SC release available on Bela, including Link support

Types of changes
----------------
Merged upstream  3.11 into the current bela 3.10 branch and made it compile.

- See Supercollider 3.11 release notes

Checklist
---------

- [ ] All tests are passing
- [ ] If necessary, new tests were created to address changes in PR, and tests are passing
- [ ] Updated documentation, if necessary
- [X] This PR is ready for review

Remaining Work
--------------

Its been a while since I was exposed to C++ code, so ymmv. The main obstacles was getting the toolchain working (distcc), working through the cmake changes and the refactoring of scsynth_main. Other than that it was pretty smooth and I verified briefly that link propagates tempo changes as expected.

### CODE REVIEW

**Comment by giuliomoro** [comment]
Thanks, I'd like instead to rebase Bela's commits on top of Sc's release, but I will keep your merge as a reference.

**Comment by argl** [comment]
Thats great, any chance that bela support will be included in the official supercollider code base?

**Comment by giuliomoro** [comment]
I did inquiry about that at some point and they suggested that it would need some way of automatic building and testing which I never did as I don't have any knowledge of continuous integration. Is that something you could perhaps help with?

**Comment by argl** [comment]
I dont have the faintest idea how sc development is done nowadays. I did not use it for 15 years, only recently. If we can get the next release out of the door with bela changes on top, we could get a discussion going, and I am happy to help if I can.

There is interest there for sure and it would benefit both communities.

**Comment by argl** [comment]
Ah and two things I noticed when playing around with it (dont know where else to put it):
* scsynth now binds to 127.0.0.1 by default, so the workflow described in the bela wiki with sclang running on a different machine will not work. You have to tell scsynth to bind to 0.0.0.0 with something like this: `scsynth -u 57110 -a 1024 -i 2 -o 2 -b 1026 -z 16 -R 0 -C 0 -l 16 -J 8 -K 0 -G 16 -B 0.0.0.0`
* The current implementation of ableton link is set up in a way that needs sclang and scsynth running on the same machine. Link is doing its thing in sclang, and sending timestamped OSC messages to the server. There seems to be some disagreement when the system clock of sclang and scsynth differ (when they run on different machines). I did not look into that, but when running all of it on the bela, the timing was as tight as expected, with sclang and scsynth on separate hosts not so much. There exists an scsynth-based link ugen that does link syncing on the server side that could be useful in some cases: https://github.com/byulparan/LinkUGen.

**Comment by giuliomoro** [comment]
hi @argl, I am preparing a [new branch](https://github.com/BelaPlatform/supercollider/tree/bela-merge-formatted-rebase) rebased on top of `develop`. It seems that it's doing OK, but I have to disable `ABLETON_LINK` when building it, because even though `cmake` runs fine, it fails at compile time for not finding some `asio` includes ... would you mind sharing how you managed to build with `ABLETON_LINK` support?

**Comment by argl** [comment]
Honestly, I dont rememeber doing anything special there, but my recollection is hazy. ASIO dependencies are pulled in as a submodule for the Link sources. I still have the build tree hanging around, if you want me to check something specific there. Note that I am not a seasoned C/C++ developer, I dont know much about cmake.

**Comment by giuliomoro** [comment]
I see, maybe I had to do some recursive submodule thing. Let me try again

**Comment by giuliomoro** [comment]
`git submodule update --init --recursive` seems to have made the trick, but apparently if you run it once without `--recursive`, you have to nuke out the submodules before it can be effective when running it again.

Thanks a bunch!

so here's a release which includes the new Scope (#75) and Ableton Link, and is rebased on top of `develop`, if you want to test it:

https://github.com/BelaPlatform/supercollider/releases/tag/v3.11-bela2
